tests/unit/amo/components/TestAddonBadges.js

```diff
@@ -47,7 +48,7 @@ describe(__filename, () => {
   it('displays a recommended badge for a recommended add-on', () => {
     const addon = createInternalAddon({
       ...fakeAddon,
-      is_recommended: true,
+      promoted: { category: RECOMMENDED, apps: [CLIENT_APP_FIREFOX] },
```

Comment:
There don't seem to be any tests (including in the other test files) that deal with `apps` being anything other than Firefox.  Can we have tests that prove the behaviour if apps contains other applications?  (Or a single application that isn't Firefox, or is `[]` or undefined)

Reply:
This isn't really the place for that test, as `AddonBadges` simply relies on the value of `isRecommended` that is generated by `createInternalAddon`, but it is a good idea to have tests for the code that generates `isRecommended`.

I enhanced the test for `isRecommended` in `createInternalAddon` so that it is testing with `[android, firefox]`, `[android]`, `[firefox]`, and `[]`. I didn't add a test for `undefined` as that is not something that the API can return.